code/test/util.js

```diff
@@ -215,6 +269,123 @@ const getEOTP = async ({ seed, hseed, effectiveTime, timeOffset }) => {
   return { index, eotp }
 }
 
+// transactionExecute commits and reveals a wallet transaction
+const transactionExecute = async ({ wallet, operationType, tokenType, contractAddress, tokenId, dest, amount, data, address, randomSeed, backlinkAddresses, testTime }) => {
```

Comment:
backlinkAddresses and randomSeed should not be an argument of a general function?

Reply:
Good Point 🙏 
Not sure the best way to do this.
I've added this to a doc to keep track of these items [REFACTOR.md](https://github.com/1wallet-dev/1wallet/blob/testing/code/test/REFACTOR.md)

Reply:
If I understand this correctly 
`ONEWallet.sol` calls `Executor.sol` using
`            Executor.execute(op, tokenTrackerState, backlinkAddresses, signatures, spendingState);`
but only receives in the OperationParameters in the reveal call
**The rest come from the ONEWallets state**
e.g.
`    IONEWallet[] backlinkAddresses; // to be set in next version - these are addresses forwarding funds and tokens to this contract AND must have their forwardAddress updated if this contract's forwardAddress is set or updated. One example of such an address is a previous version of the wallet "upgrading" to a new version. See more at https://github.com/polymorpher/one-wallet/issues/78`
So we should only ever pass Operation Paramaeters 
Even in `commitParams` and `revealParams`